src/js/main.js

```diff
@@ -2,27 +2,57 @@ import spreadsheet from './spreadsheet';
 import userColsAndRows from './helpers/userColsAndRows';
 import toggleDarkMode from './darkModeToggle/toggleDarkMode.mjs';
 import { addCellTargetingEvents } from './spreadsheet/cellNavigation';
-import { initDB } from './spreadsheet/db.js';
+import { initDB, saveCellValue, getCellValue } from './spreadsheet/db.js';
 
 const spreadsheetContainer = document.querySelector('#spreadsheetContainer');
 
-// indexedDB
+// Function to load initial data from IndexedDB
+async function loadInitialData(cols, rows) {
+  const data = [];
+  for (let row = 0; row < rows; row++) {
+    const rowData = [];
+    for (let col = 0; col < cols; col++) {
+      const cellId = `cell-${row}-${col}`;
+      const cellValue = await getCellValue(cellId);
+      rowData.push(cellValue !== null ? cellValue : '');
+    }
+    data.push(rowData);
+  }
+  return data;
+}
+
+// Initialize the spreadsheet with data from IndexedDB or default data
 initDB()
-  .then(() => {
+  .then(async () => {
     console.log('IndexedDB initialized');
 
     // DarkMode
     toggleDarkMode();
 
     const [cols, rows] = userColsAndRows();
 
+    // Load initial data
+    const initialData = await loadInitialData(cols, rows);
+
     // Create and append the spreadsheet to the container
-    spreadsheetContainer.append(spreadsheet(cols, rows));
+    const hot = spreadsheet(cols, rows, initialData);
+    spreadsheetContainer.append(hot);
 
     addCellTargetingEvents('#spreadsheetContainer table', (col, row, value) => {
-      // Here you can put the save function with params for cell contents.
-      console.log('save', col, row, value);
+      const cellId = `cell-${row}-${col}`;
+      saveCellValue(cellId, value);
     });
+
+    // Auto-save every 10 seconds
+    setInterval(() => {
```

Comment:
as discussed, I'm not sure they meant to save the whole spreadsheet every 10 seconds, or just the cell that is currently editing

Reply:
@WeronikaMartinsen what is you take on this? I don't see any reason not to save the state of the whole spreadsheet every 10 seconds. In the future the code could be extended to save more things on the page if needed as well. I just interpreted "Save state" as saving the state of the whole spreadsheet and not a single cell